kbase-extension/static/kbase/js/util/string.js

```diff
@@ -149,6 +149,31 @@ define([], () => {
         return false;
     }
 
+    /**
+     * This brings sanity to the output object name string in a few steps:
+     * 1. the name is treated as a path, and reduced down to only the filename
+     *   e.g. /path/to/file.txt -> file.txt
+     * 2. any illegal characters are transformed into _
+     *   e.g. "bad_file name&?.txt" -> "bad_file_name__.txt"
```

Comment:
if the whole name is composed of illegal characters, is that OK? i.e. the output would be `_____`

Reply:
Yeah, it's a little silly, but a valid workspace object name
![Screen Shot 2021-12-14 at 10 44 42 AM](https://user-images.githubusercontent.com/2152243/146031076-f174eb0a-e0b0-4b00-874c-c1f813a88f2e.png)
.